.github/workflows/pull-request.yaml

```
@@ -1,11 +1,212 @@
+            build/js \
+            conformance/js/failing_tests.txt \
+            conformance/js/run.sh \
+            conformance/native/build/bin/linux/conformanceReleaseExecutable/conformance.kexe \
+            conformance/native/failing_tests.txt
+
+      - name: Upload files necessary to run conformance tests
+        uses: actions/upload-artifact@v2
+        with:
+          name: conformance-test-files
+          path: conformance-test-files.tar
+
+      - name: Bundle the build report
+        if: failure()
+        run: find . -type d -name 'reports' | zip -@ -r build-reports.zip
+
+      - name: Upload the build report
+        if: failure()
+        uses: actions/upload-artifact@v2
+        with:
+          name: error-report
+          path: build-reports.zip
+
+  conformance:
+    needs: build
+    strategy:
+      matrix:
+        platform: [jvm, js, linux]
+      fail-fast: false
+    runs-on: ubuntu-18.04
+
+    steps:
+      - name: Install Java
+        if: matrix.platform == 'jvm'
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Install Node.js
+        if: matrix.platform == 'js'
+        uses: actions/setup-node@v1
+        with:
+          node-version: 14
+
+      - name: Cache protobuf
+        uses: actions/cache@v2
+        with:
+          path: ~/protobuf
+          key: ${{ runner.os }}-protobuf-3.10.1
+
+      - name: Download conformance test files
+        uses: actions/download-artifact@v2
+        with:
+          name: conformance-test-files
+
+      - name: Extract conformance test files
+        run: tar -xvf conformance-test-files.tar
+
+      - name: Run ${{ matrix.platform }} conformance tests
+        run: |
+          export CONF_TEST_PATH=~/protobuf/conformance/conformance-test-runner
+          ./conformance/test-conformance.sh ${{ matrix.platform }}
+
+  build_examples:
+    needs: build
+    runs-on: ubuntu-18.04
+
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+
+      - name: Validate Gradle Wrapper
+        uses: gradle/wrapper-validation-action@v1
+
+      - name: Download maven artifacts
+        uses: actions/download-artifact@v2
+        with:
+          name: maven-artifacts
+          path: ~/.m2/repository
+
+      - name: Cache gradle
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-examples-${{ hashFiles('examples/**/*.gradle*') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-examples-
+
+      - name: Install Java 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Build all examples
+        run: |
+          for d in examples/*; do
+            set -e
+            [ -d "$d" ] || continue
+            echo "== $d =="
+            pushd "$d" >/dev/null
+            ../../gradlew build
+            popd >/dev/null
```

Like the official error message above, it might be helpful to use the built in grouping.

```suggestion
            echo "::group::Build $d"
            pushd "$d" >/dev/null
            ../../gradlew build
            popd >/dev/null
            echo "::endgroup::"
```

---

Awesome